### PR TITLE
Update to 1.x.x Documentation

### PR DESCRIPTION
**Pull request by jeffk-kiva**
Summary of Changes
* Add documentation on how to migrate to the newest version of protocol-common.
* Also, remove CI steps that execute against node versions < 16

Signed-off-by: Jeff Kennedy <jeffk@kiva.org>

### CODE REVIEW

File: docs/migrate.to.v1.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Migrating from 0.x.x to 1.x.x
+
+There are several significant changes from 0.x.x to 1.x.x:
+* Breaking: Protocol-Common is now published as an ESModule library.
+* Breaking: Minimum node version supported is node 16.
+* Breaking: The underlying logging mechanism has been swapped out for Pino logger.
```
Comment (natesulat) [inline docs/migrate.to.v1.md:6]: Might make sense to actually specify which package has been removed, too

**Review by natesulat — APPROVED** [review]
None of this looks wrong, so 👍 

Is it worth adding a section about Jest support?

**Comment by jeffk-kiva** [comment]
> Is it worth adding a section about Jest support?

Yeah, I was thinking about this. I feel like Jest ESM support is confusing and it's own whole thing. It seems to be a little different for each project too. For example, I don't think I can just say to have set the `extensionsToTreatAsEsm` key to `[".ts"]` because that caused additional errors in `aries-controller`, but fixed other errors in `aries-guardianship-agency`. But I'm down to expand this in whatever way seems most helpful.

File: README.md [hunk 2]
```diff
@@ -10,18 +10,12 @@ npm package for your work.
...
 If you'd like to use a local version of protocol-common instead of the version available through npm packages, we've
 added some handy scripts to help you do so.
-1. In protocol-common, run the command `npm run setup`
-    - This will compile your code into a dist/ directory and use the [npm link](https://docs.npmjs.com/cli/v8/commands/npm-link)
-      utility to create a local symlink to that dist/ directory.
-2. In the project that will use protocol-common, run the command `npm link protocol-common`.
+1. In protocol-common, run the command `npm run build:pack`
+    - This will compile your code into a dist/ directory and create a `.tgz` installable binary of protocol-common.
+    - Note the path to the `.tgz`, this will be important for the next step!
+2. In the project that will use protocol-common, run the command `npm install <path to .tgz>`.
```
[the remark this answers is not in the document]
Comment (jeffk-kiva) [reply README.md:16]: Me too. It made testing the changes here a heck of a lot easier.

File: docs/migrate.to.v1.md (hunk 1 above)
Comment (jeffk-kiva) [reply docs/migrate.to.v1.md:6]: Good idea.

**Comment by natesulat** [comment]
It's more art than science, for sure, and it's at least partly dependent (heh) on the application dependencies, anyways.

Still, there are a few patterns that I've seen emerge which might be useful to add, even if they're not 100% effective. On the other hand, yes, it's a whole other thing, so it might be out of scope for changes to this package.

**Pull request merged**